code/modules/client/_client_data.dm

```diff
@@ -0,0 +1,75 @@
+// client is deleted upon logout, but we want to keep some data
+/datum/client_data
+	// just a placeholder variable for vv editor because we want to read ckey name first
+	var/aa0_hint_ckey
+	var/ckey
+
+	// simple variable to tell if they're playing
+	var/client/client
```

Comment:
This is unnecessary and risks hard dels due to how tricky it is to manage client references

Reply:
this should exist to connect to the original player, and /logout of a client is supposed to nullify this variable.
this is because all data should be visited first through /client_data thrn checks if the player is currently logged in or disconnected, and determines if it should write data into /client or /client_data, or takes the data from those.